Source/WebCore/html/parser/HTMLTreeBuilder.cpp

```diff
@@ -900,10 +901,21 @@ bool HTMLTreeBuilder::processTemplateEndTag(AtomHTMLToken&& token)
     m_tree.generateImpliedEndTags();
     if (m_tree.currentStackItem().elementName() != HTML::template_)
         parseError(token);
-    m_tree.openElements().popUntilPopped(HTML::template_);
+    m_tree.openElements().popUntil(HTML::template_);
+    RefPtr templateElement = dynamicDowncast<HTMLTemplateElement>(m_tree.openElements().top());
```

Comment:
So this code does a release type check here and then a release null check on the next line. I think the following would be slightly more efficient:
```
RELEASE_ASSERT(is<HTMLTemplateElement>(m_tree.openElements().top()); // Release type check.
Ref templateElement = downcast<HTMLTemplateElement>(*m_tree.openElements().top()); // no release check.
```

Reply:
Sure. Will do